### PR TITLE
Allow gather sources from images pulled from quay.io

### PR DESCRIPTION
**Pull request by tkdchen**
[STONEBLD-1831](https://issues.redhat.com//browse/STONEBLD-1831)

### CODE REVIEW

File: task/source-build/0.1/source-build.yaml [hunk 1]
```diff
@@ -80,6 +80,7 @@ spec:
         registry_allowlist="
         registry.access.redhat.com
         registry.redhat.io
+        quay.io
```
Comment (mmorhun) [inline task/source-build/0.1/source-build.yaml:83]: Does it make sense to limit it to specific org(s) ?

Comment (chmeliik) [reply task/source-build/0.1/source-build.yaml:83]: I think users can technically configure the build pipeline to push to any Quay organization they want, no?

Comment (tkdchen) [reply task/source-build/0.1/source-build.yaml:83]: I opened https://github.com/redhat-appstudio/build-tasks-dockerfiles/pull/70, which may be helpful for configuring this list in a more flexible way.

**Pull request closed without merging**